### ISSUE TITLE
Modal dialog changes its owner on multi screen environment

### ISSUE THREAD

**Issue by nirajdoshi1985** [Caliburn-Micro/Caliburn.Micro#305]
We have an application running in a multi-screen environment, call it **MainScreen**, **ScreenA** and **ScreenB**.

We have a user control defined as below:

```
<UserControl>    
   <ContentControl Content="{Binding ContentViewModel}"/> 
</UserControl>
```

The user control is launched on **ScreenA** with `WindowManager.ShowDialog()`:

```
 dynamic windowSettings = new ExpandoObject();
 windowSettings.WindowStartupLocation = WindowStartupLocation.CenterOwner;
 windowSettings.Owner = _screenAWindow;
 WindowManager.ShowDialog(userControl, null, windowSettings);
```

and the `WindowManager ` displays the dialog on correct screen, i.e. **ScreenA**. Now when dialog is still open on **ScreenA** and if the Content of ContentControl is changed (ContentViewModel), the dialog switches to the **MainScreen**. I don't know how to force the dialog to stay on whatever screen it was opened.

I also tried setting the `WindowStartupLocation` to `CenterScreen` and removing the `Owner`. All the screens have their own button to open that dialog. When the dialog pops up it locates the parent screen correctly i.e. opens up on whatever screen the button was clicked. But again, the problem occurs if the content of **ContentControl** is changed, it switches to **MainScreen**.

Thank you.

**Comment by nigel-sampson** [comment]
I haven't been able to recreate this. can you provide a small project that shows off the bug?

**Comment by nirajdoshi1985** [comment]
Hi Niegel,
After your reply, I built a sample project, but found that everything was working fine on this project. Then I went back to main project to dig deeper the cause of issue.

The problem was all the dialogs in our solution were inherited from a styled Window, which had following code in code-behind.
```
  SizeChanged += (sender, args) =>
  {
                var curApp = Application.Current;
                var mainWindow = curApp.MainWindow;
                Left = mainWindow.Left + (mainWindow.Width - ActualWidth) / 2;
                Top = mainWindow.Top + (mainWindow.Height - ActualHeight) / 2;
   };
```

These statements forced the UI to set their center on Main Screen if Size changed, and of course changing the content (viewmodel itself) changed the view and in-turn the size of dialog.

Sorry for the trouble and thank your for your time and reply.
